lib/matplotlib/text.py

```diff
@@ -1410,7 +1410,9 @@ def __call__(self, renderer):
         elif isinstance(self._artist, Transform):
             x, y = self._artist.transform(self._ref_coord)
         else:
-            raise RuntimeError("unknown type")
+            raise RuntimeError(
+                f"Unexpected type for 'artist'. This must be one of Artist, BboxBase "
+                f"or Transform, but got {type(self._artist).__name__}")
```

Comment:
```suggestion
                f"or Transform, but got {type(self._artist).__name__}.")
```
Since there are two sentences, but maybe not worth the CI?

(Some more case below.)

Reply:
Is this a common policy. I know that single sentence/expression exception message don't have the trailing period. No idea what to do if there is more text.

Reply:
Ok grepped throuth the CPython codebase, and indeed multi-sentence exceptions end in a period.